gdal/ogr/ogrsf_frmts/amigocloud/ogramigoclouddatasource.cpp

```diff
@@ -356,12 +357,30 @@ OGRLayer   *OGRAmigoCloudDataSource::ICreateLayer( const char *pszNameIn,
         return NULL;
     }
 
-/* -------------------------------------------------------------------- */
-/*      Do we already have this layer?  If so, should we blow it        */
-/*      away?                                                           */
-/* -------------------------------------------------------------------- */
     CPLString osName(pszNameIn);
 
+    if (nLayers > 1) {
+        // We cannot have more then one destination dataset (layer).
+        CPLError( CE_Failure, CPLE_AppDefined,
+                  "Multiple destination datasets are not allowed.");
+        return NULL;
+    } else if (nLayers == 1) {
+        // If layer already exists, and OVERWRITE: YES, truncate the layer.
+        if( CSLFetchNameValue( papszOptions, "OVERWRITE" ) != NULL  &&
+            !EQUAL(CSLFetchNameValue(papszOptions,"OVERWRITE"),"NO") )
+        {
+            TruncateDataset(papoLayers[0]->GetTableName());
+        }
+        else
+        {
+            CPLError( CE_Warning, CPLE_AppDefined,
```

Comment:
shouldn't that return NULL in that case ?

Reply:
No, we want to use this existing layer. I might just remove that warning message, it's not really necessary.